drivers/i2s/i2s_nrfx.c

```diff
@@ -0,0 +1,1310 @@
+		return;
+	}
+	if (next_buffers_needed(status)) {
+		ret = k_mem_slab_alloc(ch_rx->mem_slab,
+				       (void **)&p_new_buffers->p_rx_buffer,
+				       K_NO_WAIT);
+		if (ret < 0) {
+			/*overrun error occurred*/
+			LOG_ERROR;
+			interface_error_service();
+			channel_error_service(ch_rx);
+			return;
+		}
+	}
+}
+
+static int channel_rx_get_data(u32_t **buf, size_t *block_size)
+{
+	struct channel_str *ch_rx = get_interface()->channel_rx;
+	int ret = queue_add(&ch_rx->mem_block_queue, *buf, *block_size);
+
+	if (ret < 0) {
+		return ret;
+	}
+	k_sem_give(&ch_rx->sem);
+	return 0;
+}
+
+static void isr(void *arg)
+{
+	struct i2s_nrfx_interface *i2s = get_interface();
+
+	nrfx_i2s_irq_handler();
+	if (i2s->get_state() == I2S_IF_RESTARTING) {
+		int ret = i2s->start();
+
+		if (ret < 0) {
+			i2s->error_service();
+		}
+		return;
+	}
+}
+
+
+#ifdef CONFIG_NRFX_I2S
+struct queue_item rx_1_ring_buf[CONFIG_NRFX_I2S_RX_BLOCK_COUNT + 1];
+struct queue_item tx_1_ring_buf[CONFIG_NRFX_I2S_TX_BLOCK_COUNT + 1];
+static void setup_instance_0(struct device *dev)
+{
+	struct i2s_nrfx_data *const dev_data = DEV_DATA(dev);
+
+	queue_init(&dev_data->channel_tx.mem_block_queue,
+		   CONFIG_NRFX_I2S_TX_BLOCK_COUNT + 1, &tx_1_ring_buf[0]);
+	queue_init(&dev_data->channel_rx.mem_block_queue,
+		   CONFIG_NRFX_I2S_RX_BLOCK_COUNT + 1, &rx_1_ring_buf[0]);
+	IRQ_CONNECT(DT_INST_0_NORDIC_NRF_I2S_IRQ_0,
+		    DT_INST_0_NORDIC_NRF_I2S_IRQ_0_PRIORITY, isr, 0, 0);
+	irq_enable(DT_INST_0_NORDIC_NRF_I2S_IRQ_0);
+}
+
+static const struct i2s_nrfx_config channel_cfg_0 = {
+	.sck_pin = DT_NORDIC_NRF_I2S_I2S_0_SCK_PIN,
+	.lrck_pin =  DT_NORDIC_NRF_I2S_I2S_0_LRCK_PIN,
+	.mck_pin = DT_NORDIC_NRF_I2S_I2S_0_MCK_PIN,
+	.sdout_pin = DT_NORDIC_NRF_I2S_I2S_0_SDOUT_PIN,
+	.sdin_pin = DT_NORDIC_NRF_I2S_I2S_0_SDIN_PIN,
+	.instance_init = setup_instance_0,
+};
+
+struct i2s_nrfx_data channels_data_0 = {
+	.channel_tx = {
+		.current_state = I2S_STATE_NOT_READY,
+		.last_trigger_cmd = I2S_TRIGGER_PREPARE,
+		.start = channel_tx_start,
+		.stop = channel_tx_stop,
+		.drain = channel_tx_drain,
+		.drop = channel_tx_drop,
+		.data_handler = channel_tx_callback,
+		.get_data = channel_tx_get_data,
+		.mem_clear = channel_tx_mem_clear,
+	},
+	.channel_rx = {
+		.current_state = I2S_STATE_NOT_READY,
+		.last_trigger_cmd = I2S_TRIGGER_PREPARE,
+		.start = channel_rx_start,
+		.stop = channel_rx_stop,
+		.drain = channel_rx_stop,
+		.drop = channel_rx_drop,
+		.data_handler = channel_rx_callback,
+		.get_data = channel_rx_get_data,
+		.mem_clear = channel_rx_mem_clear,
+	},
+};
+
+static struct i2s_nrfx_interface interface = {
+	.state = I2S_IF_NOT_READY,
+	.size = 0,
+	.error_service = interface_error_service,
+	.set_state = interface_set_state,
+	.get_state = interface_get_state,
+	.start = interface_start,
+	.stop = interface_stop,
+	.restart = interface_restart,
+	.handler = interface_handler,
+};
+
+static inline struct i2s_nrfx_interface *get_interface(void)
```

Comment:
This function relies on global `interface` variable and prevents the driver from supporting more than one device instance.